icechunk/src/storage/mod.rs

```diff
@@ -323,28 +324,38 @@ where
 
 /// Split an object request into multiple byte range requests
 ///
-/// Returns tuples of Range for each request. It tries to generate the maximum number of
-/// requests possible, not generating more than `max_parts` requests, and each request not being
-/// smaller than `min_part_size`. Note that the size of the last request is >= the preceding one.
+/// Returns tuples of Range for each request.
+///
+/// It generates requests that are as similar as possible in size, this means no more than 1 byte
+/// difference between the requests.
+///
+/// It tries to generate ceil(size/ideal_req_size) requests, but never exceeds max_requests.
+///
+/// ideal_req_size and max_requests must be > 0
 fn split_in_multiple_requests(
     range: &Range<u64>,
-    min_part_size: u64,
-    max_parts: u16,
+    ideal_req_size: u64,
```

Comment:
Don't we need to rename the config parameter too?

Reply:
done